### PR TITLE
Adding biplot to the PCADecomposition visualizer

### PR DESCRIPTION
**Pull request by RaulPL**
I added a new parameter called proj_features to the PCADecomposition visualizer, now you can plot the features in projected space. With this change the visualizer will behave like a biplot. This solves issue #455 . The output is like this:
image

There is one drawback, currently I don't know how to plot arrows in 3d with matplolib so I just plot lines instead of arrows when proj_dim is equal to 3.

I will be happy to incorporate any feedback.

### CODE REVIEW

**Comment by lwgray** [comment]
Can you post an example visualization?

**Comment by RaulPL** [comment]
I used this code to produce the image above.

```python
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.datasets import load_iris
from yellowbrick.features.pca import PCADecomposition

iris = load_iris()
df_iris = pd.DataFrame(data=iris.data, columns=iris.feature_names)
pca_viz = PCADecomposition(proj_features=True, proj_dim=2)
X = pca_viz.fit_transform(df_iris)
pca_viz.poof()
```

**Comment by bbengfort** [comment]
@RaulPL sorry about the other note; I was addressing issues from email and didn't realize there were already two PRs! I'll go ahead and take a look at this one shortly!

File: yellowbrick/features/pca.py [hunk 1]
```diff
@@ -144,25 +148,30 @@ class PCADecomposition(FeatureVisualizer):
...
+             ('pca', PCA(self.proj_dim, ))]
+        )
+        self.proj_features = proj_features
         # Visual Parameters
         self.colormap = colormap
 
     def fit(self, X, y=None, **kwargs):
+        super(PCADecomposition, self).fit(X=X, y=y, **kwargs)
         self.pca_transformer.fit(X)
+        self.pca_components_ = self.pca_transformer.steps[1][1].components_
```
Comment (bbengfort) [inline yellowbrick/features/pca.py]: Excellent, thank you for adding this to a learned `pca_components_` attribute, I think this is the exact correct place to store this information. 

In the future, I may update this to `self.pca_transformer.named_steps['pca'].components_` -- but only for understandability to new contributors; this is perfectly fine right now!

File: yellowbrick/features/pca.py [hunk 2]
```diff
@@ -172,26 +181,57 @@ def transform(self, X, y=None, **kwargs):
...
+                        [0, x_vector[i] * max_x],
+                        [0, y_vector[i] * max_y],
+                        [0, z_vector[i] * max_z],
+                        color='r'
+                    )
+                    self.ax.text(
+                        x_vector[i] * max_x * 1.05,
+                        y_vector[i] * max_y * 1.05,
+                        z_vector[i] * max_z * 1.05,
+                        self.features_[i], color='r'
```
Comment (bbengfort) [inline yellowbrick/features/pca.py]: You were saying something before about having difficulty with the 3D vectors; has that been resolved? 

Also, what is the 1.05 multiplier here? Just offsetting the text by that much from the arrow?

File: yellowbrick/features/pca.py [hunk 3]
```diff
@@ -172,26 +181,57 @@ def transform(self, X, y=None, **kwargs):
...
-            self.set_title('Principal Component Plot')
-            self.ax.set_ylabel('Principal Component 2')
-            self.ax.set_xlabel('Principal Component 1')
-
-        else:
-            self.fig.set_title('Principal Component Plot')
-            self.fig.set_xlabel('Principal Component 1')
-            self.fig.set_ylabel('Principal Component 2')
-            self.fig.set_zlabel('Principal Component 3')
+        self.ax.set_title('Principal Component Plot')
```
Comment (bbengfort) [inline yellowbrick/features/pca.py]: Thanks for cleaning this up!

File: yellowbrick/features/pca.py [hunk 4]
```diff
@@ -172,26 +181,57 @@ def transform(self, X, y=None, **kwargs):
...
-
         if self.proj_dim == 2:
             self.ax.scatter(X[:, 0], X[:, 1], c=self.color, cmap=self.colormap)
+            if self.proj_features:
+                x_vector = self.pca_components_[0]
+                y_vector = self.pca_components_[1]
+                max_x = max(X[:, 0])
+                max_y = max(X[:, 1])
+                for i in range(self.pca_components_.shape[1]):
+                    plt.arrow(
```
Comment (bbengfort) [inline yellowbrick/features/pca.py]: We prefer the `self.ax.arrow` and `self.ax.text` methods to ensure that the correct axes is drawn upon (just for future reference, we hope you'll contribute more!). I can update this though.

**Comment by RaulPL** [comment]
Hi, don't worry, I have mess with my branches. Do you think it is a good idea to delete my fork of the project and make a new fresh pull request?

**Comment by bbengfort** [comment]
@RaulPL actually, I think this PR is ready to be merged now. The question is what to do about #430 -- if you could answer my questions above, I can merge this in right now; then update #430 with the latest, if you could do a pull then that might sync everything up.

File: yellowbrick/features/pca.py (hunk 1 above)
Comment (RaulPL) [reply yellowbrick/features/pca.py]: I did not know about "named_steps", I can fix it at night.

Comment (bbengfort) [reply yellowbrick/features/pca.py]: Don't worry about it! Using steps is perfectly acceptable; I'd like to go ahead an merge this now if that's ok with you, then I can quickly add the visual tests, and we can get back on track with #430!

File: yellowbrick/features/pca.py (hunk 2 above)
Comment (RaulPL) [reply yellowbrick/features/pca.py]: I found a way to plot 3D arrows in matplolib [here](https://stackoverflow.com/questions/22867620/putting-arrowheads-on-vectors-in-matplotlibs-3d-plot)  but it requires to create a class that inherits from FancyArrowPatch, so I decided to keep it simple and just plot a line.

And yes, is to get the offset

File: yellowbrick/features/pca.py (hunk 4 above)
Comment (RaulPL) [reply yellowbrick/features/pca.py]: OK, sure I will have it in mind

Comment (bbengfort) [reply yellowbrick/features/pca.py]: Thanks!

File: yellowbrick/features/pca.py (hunk 2 above)
Comment (bbengfort) [reply yellowbrick/features/pca.py]: Awesome, thanks for clarifying!

**Comment by bbengfort** [comment]
@RaulPL Thank you so much for adding this to our PCA visualizer; it's details like this that I think make Yellowbrick truly outstanding for ML diagnostics, and this is a creative and very informative addition to the visualizer!

**Pull request merged**